sormas-backend/src/test/java/de/symeda/sormas/backend/ArchitectureTest.java

```diff
@@ -87,23 +97,69 @@ public void testClinicalVisitFacadeEjbAuthorization(JavaClasses classes) {
+	}
+
+	@ArchTest
+	public void testLabMessageFacadeEjbAuthorization(JavaClasses classes) {
+		List<String> specificMethodNames = new ArrayList<>();
+		specificMethodNames.add("fetchAndSaveExternalLabMessages");
+		assertFacadeEjbAnnotated(LabMessageFacadeEjb.class, true, classes, specificMethodNames);
+	}
+
+	@ArchTest
+	public void testTestReportFacadeEjbAuthorization(JavaClasses classes) {
+		assertFacadeEjbAnnotated(TestReportFacadeEjb.class, true, classes, null);
+	}
+
+	@ArchTest
+	public void testImmunizationFacadeEjbAuthorization(JavaClasses classes) {
+		assertFacadeEjbAnnotated(ImmunizationFacadeEjb.class, classes);
+	}
+
+	@ArchTest
+	public void testVaccinationFacadeEjbAuthorization(JavaClasses classes) {
+		assertFacadeEjbAnnotated(VaccinationFacadeEjb.class, classes);
+	}
+
+	@ArchTest
+	public void testTravelEntryFacadeEjbAuthorization(JavaClasses classes) {
+		assertFacadeEjbAnnotated(TravelEntryFacadeEjb.class, classes);
+	}
+
+	@ArchTest
+	public void testTaskFacadeEjbAuthorization(JavaClasses classes) {
+		assertFacadeEjbAnnotated(TaskFacadeEjb.class, classes);
+	}
+
 	private void assertFacadeEjbAnnotated(Class<?> facadeEjbClass, JavaClasses classes) {
-		assertFacadeEjbAnnotated(facadeEjbClass, false, classes);
+		assertFacadeEjbAnnotated(facadeEjbClass, false, classes, null);
 	}
 
-	private void assertFacadeEjbAnnotated(Class<?> facadeEjbClass, boolean classAuthOnly, JavaClasses classes) {
+	private void assertFacadeEjbAnnotated(Class<?> facadeEjbClass, boolean classAuthOnly, JavaClasses classes, List<String> specificMethodNames) {
 		ArchRuleDefinition.theClass(facadeEjbClass).should().beAnnotatedWith(RolesAllowed.class).check(classes);
-
-		GivenMethodsConjunction methods = ArchRuleDefinition.methods().that().areDeclaredIn(facadeEjbClass).and().arePublic();
+		GivenMethodsConjunction allPublicMethods = ArchRuleDefinition.methods().that().areDeclaredIn(facadeEjbClass).and().arePublic();
 
 		if (classAuthOnly) {
-			methods.should().notBeAnnotatedWith(RolesAllowed.class);
+			getMethodsShouldNotBeAnnotated(facadeEjbClass, allPublicMethods).should().notBeAnnotatedWith(RolesAllowed.class);
+			if (specificMethodNames != null) {
+				specificMethodNames.forEach(
+					specificMethodName -> allPublicMethods.and().haveFullName(specificMethodName).should().beAnnotatedWith(RolesAllowed.class));
+			}
 		} else {
-			methods.should()
+			allPublicMethods.should()
 				.beAnnotatedWith(RolesAllowed.class)
 				.orShould()
-				.haveNameMatching("^(get|count|is|does|has|validate|to|pseudonymize|convertToReferenceDto|fillOrBuild|convertToDto|fromDto).*")
+				.haveNameMatching("^(get|count|is|does|has|validate|to|pseudonymize|convertToReferenceDto|fillOrBuild|convertToDto|fromDto|exists).*")
 				.check(classes);
 		}
 	}
+
+	private GivenMethodsConjunction getMethodsShouldNotBeAnnotated(Class<?> facadeEjbClass, GivenMethodsConjunction methods) {
+		return !facadeEjbClass.getSimpleName().equals("LabMessageFacadeEjb")
```

Comment:
this should be done using the `specificMethodNames` from `assertFacadeEjbAnnotated` instead of hardcoded class name and method name